newIDE/app/src/Course/LockedCourseChapterPreview.js

```diff
@@ -0,0 +1,218 @@
+    <Lock style={styles.lockerImage} />
+  </div>
+);
+
+type Props = {|
+  courseChapter: LockedVideoBasedCourseChapter | LockedTextBasedCourseChapter,
+  onBuyWithCredits: (CourseChapter, string) => Promise<void>,
+|};
+
+const LockedCourseChapterPreview = React.forwardRef<Props, HTMLDivElement>(
+  ({ courseChapter, onBuyWithCredits }, ref) => {
+    const { openSubscriptionDialog } = React.useContext(
+      SubscriptionSuggestionContext
+    );
+    const [
+      displayPasswordPrompt,
+      setDisplayPasswordPrompt,
+    ] = React.useState<boolean>(false);
+    const { windowSize } = useResponsiveWindowSize();
+    const [password, setPassword] = React.useState<string>('');
+    const youtubeVideoId = courseChapter.videoUrl
+      ? getYoutubeVideoIdFromUrl(courseChapter.videoUrl)
+      : null;
+    const [isPurchasing, setIsPurchasing] = React.useState<boolean>(false);
+
+    const onClickBuyWithCredits = React.useCallback(
+      async () => {
+        if (!courseChapter.isLocked) return;
+
+        setDisplayPasswordPrompt(false);
+        setIsPurchasing(true);
+        try {
+          await onBuyWithCredits(courseChapter, password);
+        } finally {
+          setIsPurchasing(false);
+        }
+      },
+      [courseChapter, onBuyWithCredits, password]
+    );
+
+    const onWillBuyWithCredits = React.useCallback(
+      async () => {
+        // Password is required in dev environment only so that one cannot freely claim asset packs.
+        if (Window.isDev()) setDisplayPasswordPrompt(true);
+        else onClickBuyWithCredits();
+      },
+      [onClickBuyWithCredits]
+    );
+
+    return (
+      <div style={styles.videoAndMaterialsContainer}>
+        {youtubeVideoId && (
+          <div
+            style={{
+              ...styles.videoContainer,
+              maxWidth: windowSize === 'xlarge' ? 960 : 640,
+            }}
+          >
+            <img
+              alt={`Video for lesson ${courseChapter.title}`}
+              style={styles.videoThumbnail}
+              src={`https://i.ytimg.com/vi/${youtubeVideoId}/sddefault.jpg`}
+            />
+            <LockedOverlay />
+          </div>
+        )}
+        <div style={styles.lockedChapterTextContainer}>
+          <Paper background="medium" style={styles.sideBar}>
+            <ColumnStackLayout noMargin justifyContent="center">
+              <Text noMargin size="sub-title">
+                <Trans>Unlock this lesson to finish the course</Trans>
+              </Text>
+              <Text noMargin>
+                <Trans>
+                  Use your GDevelop credits to purchase lessons in this course;
+                  or get a subscription to get them for free.
+                </Trans>
+              </Text>
+              <ResponsiveLineStackLayout
+                noMargin
+                noColumnMargin
+                forceMobileLayout={
+                  windowSize === 'small' || windowSize === 'medium'
```

Comment:
probably don't need the check on windowSize small here, as it will do it automatically